romeo/src/state.rs

```diff
@@ -125,50 +151,105 @@ pub fn update(config: &Config, state: State, event: Event) -> (State, Vec<Task>)
         }
         Event::BitcoinBlock(block) => process_bitcoin_block(config, state, block),
         Event::AssetContractCreated(txid) => process_asset_contract_created(config, state, txid),
+        Event::MintCreated(deposit_info, txid) => process_mint_created(state, deposit_info, txid),
         event => panic!("Cannot handle yet: {:?}", event),
     }
 }
 
-fn process_bitcoin_block(_config: &Config, mut state: State, block: Block) -> (State, Vec<Task>) {
-    let deposits = parse_deposits(&block);
+fn process_bitcoin_block(config: &Config, mut state: State, block: Block) -> (State, Vec<Task>) {
+    let deposits = parse_deposits(config, &block);
     let withdrawals = parse_withdrawals(&block);
 
     state.deposits.extend_from_slice(&deposits);
     state.withdrawals.extend_from_slice(&withdrawals);
 
-    state.block_height = block
+    let new_block_height = block
```

Comment:
Does it also work to do this?

```
let new_block_height: u32 = block
        .bip34_block_height()
        .expect("Failed to get block height");
```

If this is valid it would be better to explicitly type the variable as opposed to typing the output and implicitly typing the variable.

Reply:
I don't think that will work. The cast needs to be explicit.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3241a4156cd68f163d6053b829a9bb7f